src/backend/distributed/commands/utility_hook.c

```diff
@@ -662,6 +673,27 @@ ProcessUtilityInternal(PlannedStmt *pstmt,
 				/* no failures during CONCURRENTLY, mark the index as valid */
 				MarkIndexValid(indexStmt);
 			}
+
+			/*
+			 * We make sure schema name is not null in the PreprocessIndexStmt.
+			 */
+			Oid schemaId = get_namespace_oid(indexStmt->relation->schemaname, true);
+			Oid relationId = get_relname_relid(indexStmt->relation->relname, schemaId);
+
+			/*
+			 * If this a partitioned table, and CREATE INDEX was not run with ONLY,
+			 * we have wrong index names generated on indexes of shards of
+			 * partitions of this table, so we should fix them.
+			 */
+			if (IsCitusTable(relationId) && PartitionedTable(relationId) &&
+				indexStmt->relation->inh)
+			{
+				/* only fix this specific index */
+				Oid indexRelationId =
+					get_relname_relid(indexStmt->idxname, schemaId);
+
```

Comment:
One another idea could have been to call `FixPartitionShardIndexNames` in a single place such that we have a single function checks whether we should call it or not. But, that seems to make it harder to follow, so skipped pushing that commit.